src/classLibrary/ParsingStrategies/IfElseResolver.ts

```diff
@@ -0,0 +1,210 @@
+                    } else {
+                        result.set(flag, count);
+                    }
+                });
+            });
+        });
+
+        return result;
+  	}
+
+	private CountKeyInVFS(key: string, virtualFile: VirtualFileSystemInstance): number {
+		return VirtualFileSystemInstance.match(virtualFile, {
+			File: (file: FileSystemInstance) => {
+				let tempCount = 0;
+				if (file["content"] != null && file.ignore.ifElseResolver.content) {
+					FileContent.if.String(file.content, (str) => {
+						tempCount += str.Count(key);
+					});
+				}
+				return tempCount;
+			},
+			default: () => 0,
+		});
+	}
+
+	ModifyIfWithAllSyntax(v: string, syntaxes: Syntax[]): string[] {
+		const allPossibleIfs: string[] = [];
+		syntaxes.Each(syntax => {
+			allPossibleIfs.push(`if${syntax[0] + v + syntax[1]}`);
+		})
+		return allPossibleIfs;
+	}
+
+	ModifyInverseIfWithAllSyntax(v: string, syntaxes: Syntax[]): string[] {
+		const allPossibleInverseIfs: string[] = [];
+		syntaxes.Each(syntax => {
+			allPossibleInverseIfs.push(`if!${syntax[0] + v + syntax[1]}`);
+		})
+		return allPossibleInverseIfs;
+	}
+
+	ConvertIfToEndKeyword(ifStatement: string): string
+	{
+		return ifStatement.replace('if', 'end');
+	}
+
+	
+	ResolveFiles(cyan: CyanSafe, virtualFiles: VirtualFileSystemInstance[]): VirtualFileSystemInstance[] {
+		return virtualFiles;
+	};
+
+	ResolveContents(cyan: CyanSafe, virtualFiles: VirtualFileSystemInstance[]): VirtualFileSystemInstance[] {
+		const flagsMap: Map<string, boolean> = this.util.FlattenObject(cyan.flags).MapValue((boolString: string) => boolString == "true");
+        const allPossibleIfSignaturesMap: Map<string[], boolean> = flagsMap.MapKey((key: string) => this.ModifyIfWithAllSyntax(key, cyan.syntax));
+		const allPossibleInverseIfSignaturesMap: Map<string[], boolean> = flagsMap.MapKey((key: string) => this.ModifyInverseIfWithAllSyntax(key, cyan.syntax));
+        return virtualFiles.Each((virtualFile: VirtualFileSystemInstance) => {
+            VirtualFileSystemInstance.match(virtualFile, {
+                File: (file: FileSystemInstance) => {
+                    if (!file.ignore.ifElseResolver.content) return;
+                    if (file["content"] == null) return;
+					
+                    FileContent.if.String(file.content, (strContent: string) => {
+                        allPossibleIfSignaturesMap.Each((ifSignatures: string[], v: boolean) => {
+							ifSignatures.Each((ifSignature: string) => {
+								let startIndexes: number[] = this.RetrieveLineIndexContainingSyntax(strContent, ifSignature);
+								let endIndexes: number[] = this.RetrieveLineIndexContainingSyntax(strContent, 
+									this.ConvertIfToEndKeyword(ifSignature));
+								if (v) {
+									strContent = this.RemoveLineIndexes(startIndexes.concat(endIndexes), strContent);
+								} else {
+									strContent = this.RemoveContentBetweenLineIndexes(startIndexes, endIndexes, strContent);
+								}	
```

Comment:
This part seems rather similar to lines 106 - 115. Maybe can abstract this out to become a separate function also?